Adding CONTRIBUTING.md file (#11)
SamDev98/password-manager-py: CONTRIBUTING.md

# Contributing

When contributing to this repository, please first discuss the change you wish to make via issues on the repo, with the
owners of this repository before making a change.

Please note we have a [code of conduct](CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.

## Pull Request Process

1. Ensure all new requirements are added to requirements.txt in the respective project directories.
2. Update the [README.md](README.md) with details of changes to the application.
3. Once a pull request is approved by the code owners, a maintainer will merge it.

## Setup For Development:

1. For tkinter-gui, the dev setup process is the same as for installing it.

> 2. It can be found in the [README](Tk_README.md) for the tkinter-based application.

Reviewer: Instead of just the `README` I think you can point to a specific section of the readme as `README.MD#section-name` for the installation process.  

A personal view, but you can create a separate section for installation and running the application instead of having it under documentation or maybe give it a heading just to make it more visible and also include the step installing all dependencies using `pip install requirements.txt` because not everyone will not know that

Author: Basically, all of the steps in Tk_README have to be followed regardless of the fact if this is to set up as a user application or for a developer, so I haven't tagged a particular header.

About the other point, do you mean I should give specific commands in the readme? Since I have linked the docs where they explain the procedure to install it.

# Contributing

When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other

> method with the owners of this repository before making a change.

Reviewer 2: I recommend removing the email. Keep the discussions scoped to the repository so other contributors are not excluded.

Author: Okay, please check my latest commit, I hope i have resolved these correctly.

> Please note we have a code of conduct, please follow it in all your interactions with the project.

Reviewer 2: See my comment below. This should link to the code of conduct file.

## Pull Request Process

1. Ensure all new requirements are added to requirements.txt in the respective project directories.
2. Update the [README.md](README.md) with details of changes to the application.

> 3. You may merge the Pull Request in once you have the sign-off of the code-owner.

Reviewer 2: So anyone can merge a pull request?

```suggestion
3. Once a pull request is approved by the code owners, a maintainer will merge it.
```

address, posting via an official social media account, or acting as an appointed representative at an online or offline
event. Representation of a project may be further defined and clarified by project maintainers.

### Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team
at [this email](mailto:rakhesambhav@gmail.com). All complaints will be reviewed and investigated and will result in a
response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain
confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be
posted separately.

Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent
repercussions as determined by other members of the project's leadership.

### Attribution

This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 2.0, available
at [http://contributor-covenant.org/version/2/0][version]

[homepage]: http://contributor-covenant.org

> [version]: http://contributor-covenant.org/version/2/0/

Reviewer 2: This should go in a `CODE_OF_CONDUCT.md` file, separate from your contributing doc.
